From afa2aa6a3cef7e565fd0f58701208280ebba7137 Mon Sep 17 00:00:00 2001
From: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
Date: Wed, 26 Apr 2017 13:53:08 -0500
Subject: [PATCH 2030/2094] drm/amdgpu: Use GPU PAGE SIZE to decide GART size

Change-Id: Ia25e87511dfb19b263f8d3cb6cf140096cb58a24
Signed-off-by: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>

 Conflicts:
	drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
	drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
---
 drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c | 8 ++++++++
 drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c | 8 ++++++++
 drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c | 3 ++-
 3 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
index 3665ac3..d8a6286 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v7_0.c
@@ -397,6 +397,14 @@ static int gmc_v7_0_mc_init(struct amdgpu_device *adev)
 
 	/* Unless the user has overridden it, compute the GART size */
 	if (amdgpu_gart_size == -1) {
+		/* Maximum GTT size is limited by the GART table size
+		 * in visible VRAM and the address space. Use at most
+		 * half of each. */
+		uint64_t max_gtt_size = min(
+			adev->mc.visible_vram_size / 8 *
+				AMDGPU_GPU_PAGE_SIZE / 2,
+			1ULL << 39);
+
 		si_meminfo(&si);
 		/* Set the GART to map the largest size between either
 		 * VRAM capacity or double the available physical RAM,
diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
index 4af6782..8ca1f4c 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
@@ -563,6 +563,14 @@ static int gmc_v8_0_mc_init(struct amdgpu_device *adev)
 
 	/* Unless the user has overridden it, compute the GART size */
 	if (amdgpu_gart_size == -1) {
+		/* Maximum GTT size is limited by the GART table size
+		 * in visible VRAM and the address space. Use at most
+		 * half of each. */
+		uint64_t max_gtt_size = min(
+			adev->mc.visible_vram_size / 8 *
+				AMDGPU_GPU_PAGE_SIZE / 2,
+			1ULL << 39);
+
 		si_meminfo(&si);
 		/* Set the GART to map the largest size between either
 		 * VRAM capacity or double the available physical RAM,
diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
index a9a7a14..21e7c42 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v9_0.c
@@ -504,7 +504,8 @@ static int gmc_v9_0_mc_init(struct amdgpu_device *adev)
 		 * half of each.
 		 */
 		uint64_t max_gtt_size = min(
-			adev->mc.visible_vram_size / 8 * PAGE_SIZE / 2,
+			adev->mc.visible_vram_size / 8 *
+				AMDGPU_GPU_PAGE_SIZE / 2,
 			1ULL << 39);
 
 		si_meminfo(&si);
-- 
2.7.4

